### PR TITLE
Get using-mongodb example working on netlify

### PR DESCRIPTION
**Pull request by jorishermans**
- Get ride of the error to bootstrap the project

### CODE REVIEW

File: packages/gatsby/src/schema/infer-graphql-input-fields.js [hunk 1]
```diff
@@ -205,6 +205,7 @@ export function inferInputObjectStructureFromNodes({
       prefix: `${prefix}${_.upperFirst(key)}`,
     })
 
+    if (typeof key !== 'string') return
```
Comment (KyleAMathews) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: This isn't the right fix — source/transformer plugins need to ensure that they're not passing in bad data. Add a console.log here to see what bad data gatsby-source-mongodb is passing in and fix it there.

Comment (jorishermans) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: Yes that is what I am doing now ... 

I will keep you posted what he is printing out here. I can't see how this is working, this is still magic for me so need so help I suppose :)

Comment (KyleAMathews) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: :-)

Gatsby auto-creates a GraphQL schema from your data. Part of which is generating the field names / keys for the various "objects" in the schema. Keys have to be strings that start with [a-zA-Z].

Source/transformer plugins have to ensure this is the case.

Comment (jorishermans) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: keys are numbers instead of a string, how can that be

**Comment by jorishermans** [comment]
@KyleAMathews I get the following info.

building schemakey is ... 0 and field is {"type":"mongodbCloudDocumentsIdId0QueryInteger"}
{"type":"Buffer","data":[89,124,153,101,37,118,199,74,3,104,120,152]}

Any idea's?

File: packages/gatsby/src/schema/infer-graphql-input-fields.js (hunk 1 above)
Comment (KyleAMathews) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: Mongo returns numbers for keys? `console.log` out the nodes you're creating.

**Comment by KyleAMathews** [comment]
Does your nodes have arrays of arrays?

**Comment by jorishermans** [comment]
I found it :)

MongoDB item has a _id field and graphql, doesn't like that ... when I do a delete on that property of the mongoDB field it is working.

Now why do I get this error?
This dependency was not found:

* react-router-scroll in ./.cache/root.js
To install it, you can run: npm install --save react-router-scroll

**Comment by KyleAMathews** [comment]
Yeah!

>Now why do I get this error?

Probably an NPM install error.

I recommend using Yarn whenever possible.

**Comment by jorishermans** [comment]
I add the console.logs for debugging purposes, this is very handy to print out the fieldname that is bad in the data and removed the return in the infer-graphql-input-fields.js

It really scratches my head to get on this, but now it makes total sense.

I was experimenting lately with Yarn ... will give that a try.

**Comment by jorishermans** [comment]
@KyleAMathews this works after I deleted the .cache folder again. So we have clear setup and that should memic the state of netlify.

The code of the mapping functionality is been put in comment.

**Comment by jorishermans** [comment]
I will

**Comment by jorishermans** [comment]
Well the createKey has no field ...

And passing by field to createKey only for debug purpose is not a good idea. So I guess it is beter to leave it at that place.

Don't know what you think of that?

File: packages/gatsby/src/schema/infer-graphql-input-fields.js [hunk 2]
```diff
@@ -204,8 +204,13 @@ export function inferInputObjectStructureFromNodes({
       value,
       prefix: `${prefix}${_.upperFirst(key)}`,
     })
-
+ 
     if (field == null) return
+    // check if your key is really a string otherwise you have bad data ...
+    if (typeof key !== 'string') {
```
Comment (KyleAMathews) [inline packages/gatsby/src/schema/infer-graphql-input-fields.js]: This still needs moved into the create-key.js file

**Comment by KyleAMathews** [comment]
Awesome, thanks for the great work here!

**Pull request merged**